### PR TITLE
GPU implementation of the Mean Squared Error loss function

### PR DESCRIPTION
**Pull request by ghost**
This PR adds support for the Mean Squared Error loss function on GPU.
Other PRs:
dll: https://github.com/wichtounet/dll/pull/18
egblas: https://github.com/wichtounet/etl-gpu-blas/pull/1

### CODE REVIEW

File: include/etl/impl/egblas/mse.hpp [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+ * \param ldb The leading dimension of b
+ */
+inline std::pair<double, double> mse([[maybe_unused]] size_t n,
+                                     [[maybe_unused]] double alpha,
+                                     [[maybe_unused]] double beta,
+                                     [[maybe_unused]] const double* A,
+                                     [[maybe_unused]] size_t lda,
+                                     [[maybe_unused]] const double* B,
+                                     [[maybe_unused]] size_t ldb) {
+#ifdef EGBLAS_HAS_SMSE
```
Comment (wichtounet) [inline include/etl/impl/egblas/mse.hpp:221]:
```suggestion
#ifdef EGBLAS_HAS_DMSE
```

File: include/etl/impl/egblas/mse.hpp [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    cpp_unreachable("Invalid call to egblas::mse");
+
+    return std::make_pair(0.0f, 0.0f);
+#endif
+}
+
+/*!
+ * \brief Indicates if EGBLAS has double-precision MSE
+ */
+#ifdef EGBLAS_HAS_MSE_DLOSS
```
Comment (wichtounet) [inline include/etl/impl/egblas/mse.hpp:199]:
```suggestion
#ifdef EGBLAS_HAS_DMSE
```

File: include/etl/impl/egblas/mse.hpp [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+    cpp_unreachable("Invalid call to egblas::mse_error");
+
+    return 0.0;
+#endif
+}
+
+/*!
+ * \brief Indicates if EGBLAS has single-precision MSE
+ */
+#ifdef EGBLAS_HAS_MSE_SLOSS
```
Comment (wichtounet) [inline include/etl/impl/egblas/mse.hpp:164]:
```suggestion
#ifdef EGBLAS_HAS_SMSE
```

**Review by wichtounet — APPROVED** [review]
Just a few changes necessary and it will be fine 👍 

I have already fixed the corresponding flags from the wrapper. You only copied my mistakes ;)

**Comment by niciuffo** [comment]
Didn't know you could commit suggestions right from github, I pushed a commit to fix those flags though :)

**Comment by wichtounet** [comment]
> Didn't know you could commit suggestions right from github, I pushed a commit to fix those flags though :)

Yes, that's a pretty cool feature!

**Pull request merged**